src/oca_pre_commit_hooks/checks_odoo_module_po.py

```diff
@@ -106,18 +106,29 @@ def check_po_pretty_format(self):
                 filepath=self.filename_short,
             )
 
+    @staticmethod
+    def extract_line_no(error_msg):
+        """Removes (line n) from the message error and returns n with the new message"""
+        line_regex = re.search(r" \(line (\d+)\)", error_msg)
+        line_no = None
+        if line_regex:
+            line_no = int(line_regex.group(1))
+            error_msg = error_msg[: line_regex.start()] + error_msg[line_regex.end() :]
```

Comment:
What about using `re.sub` instead?

Reply:
It was just to parse the regex once. With re.sub I need to parse it once for finding the line number and once for the substitution.

Reply:
Check my latest commit if you agree

Reply:
👍

Reply:
Although I would leave `line = None` because if no line is found the line number is left empty instead of (erroneously) at 1

Reply:
That is a different change for this PR since that we are waiting a integer but not sure if we have script to process `None` from another tools

FYI I have found a way to get the match and the replace in one shot

Reply:
> FYI I have found a way to get the match and the replace in one shot

Ah right, very cool